### PR TITLE
Odbc

### PR DESCRIPTION
**Pull request by stijnvanhoey**
* example of using an existing odbc connection within RODBC
* introduction to major RODBC functions
* from qurey to template based function basic example

### CODE REVIEW

File: source/data-handling-database-R.Rmd [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+![access-dsn](../img/access_dsn.png)
+
+For example, the DSN name `UserFlora` or `Cydonia-prd` can be used to query these databases and extract data from it with similar queries to the one used in MSAccess. First of all, the connection with the database need to be established, by using the `odbcConnect` function, providing the DSN name as argument:
+
+```{r sqlconnection}
+# WINDOWS USERS (uncomment when working in Windows)
+#my_connection <- odbcConnect("UserFlora") # uncomment for windows users
+
+# LINUX/MAC USERS (comment out when working in Windows)
+my_connection <- odbcDriverConnect("Driver={ODBC Driver 13 for SQL Server};Server=inbosql04.inbo.be;Database=UserFlora;Trusted_Connection=yes;")
```
Comment (ThierryO) [inline source/data-handling-database-R.Rmd]: 1. This assumes a working LDAP. I would be nice to have some instructions to setup LDAP on Linux/Mac (should be in a separate document.
2. Does this requires other dependencies than just RODBC?

File: source/data-handling-database-R.Rmd [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+```{r queryexample}
+subset_meting <- sqlQuery(my_connection, 
+"SELECT meet.Cor_X
+	 , meet.COr_Y
+	 , meet.MetingStatusCode
+	 , tax.NaamNederlands
+	 , tax.NaamWetenschappelijk
+	 , waar.IFBLHokID
+FROM  [dbo].[tblMeting] meet
```
Comment (ThierryO) [inline source/data-handling-database-R.Rmd]: [dbo].[tblMeting] **AS** meet

I prefer to be a bit verbose as that is more readable, especially for novices.

File: source/data-handling-database-R.Rmd [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+## Create and use query templates
+
+When you regularly use similar queries, with some minimal alterations, you do not want to copy/paste each time the entire query. It is prone to errors and you're script will become verbose. It is advisable to create query *templates*, that can be used within the `sqlQuery` function. 
+
+Consider the execution of the following query. We are interested in those records with valid X and Y coordinates for the measurement, based on a given dutch name:
+
+```{r queryexample}
+subset_meting <- sqlQuery(my_connection, 
+"SELECT meet.Cor_X
+	 , meet.COr_Y
```
Comment (ThierryO) [inline source/data-handling-database-R.Rmd]: meet.C**O**r_Y or meet.C**o**r_Y?

File: source/data-handling-database-R.Rmd [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+**Remark:** Do not forget to close your connection when done finished. 
+
+```{r closeit}
+close(my_connection)
+```
+
+## The `sprintf` function {#sprintf}
+
+In order to accomplish the re-usage of a query for different input names (`dutch_name`), the `sprintf` function is used. The `sprintf` function provides the ability to combine text and variable values in a single charactor string (i.e. the query to execute). For each variable name required in the query (any part of your query you want to have interchangeable), a representation in the query is given by a `%` in combination with the data type you want to represent. The latter is an agreed format conversion, for example `%s` is a character string (e.g. the example of `dutch_name`) and `%d` is an integer value (e.g. `2`). More information about the different symbols is given in the documentation of `sprintf` (type `?sprintf` in the console).
```
Comment (ThierryO) [inline source/data-handling-database-R.Rmd]: I find `%i` more intuitive for integers.

File: source/data-handling-database-R.Rmd [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+```{r closeit}
+close(my_connection)
+```
+
+## The `sprintf` function {#sprintf}
+
+In order to accomplish the re-usage of a query for different input names (`dutch_name`), the `sprintf` function is used. The `sprintf` function provides the ability to combine text and variable values in a single charactor string (i.e. the query to execute). For each variable name required in the query (any part of your query you want to have interchangeable), a representation in the query is given by a `%` in combination with the data type you want to represent. The latter is an agreed format conversion, for example `%s` is a character string (e.g. the example of `dutch_name`) and `%d` is an integer value (e.g. `2`). More information about the different symbols is given in the documentation of `sprintf` (type `?sprintf` in the console). 
+
+```{r sprintfex1}
+aname <- 'Jan'
```
Comment (ThierryO) [inline source/data-handling-database-R.Rmd]: separate the words with underscores to make it more readable: `a_name` vs `aname`

File: source/data-handling-database-R.Rmd [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+aninteger <- 3
+afloat <- 2.8
+sprintf('This prints a combination of a name: %s, an integer: %d and a float value: %f', aname, aninteger, afloat)
+```
+
+For float values, you can further define the fixed number of decimals (`%f`)
```
Comment (ThierryO) [inline source/data-handling-database-R.Rmd]: or a fixed number of significant digits with `%g`

File: source/data-handling-database-R.Rmd (hunk 1 above)
Comment (stijnvanhoey) [inline source/data-handling-database-R.Rmd]: I agree, whereas this is indeed out of scope for this tutorial, it should be provided elsewhere. I'll create a separate document for linux-users in the near future, as this requires an ODBC-driver and kerberos utilities for linux-users. 

I focused with this tutorial on windows-users with an existing ODBC connection.

File: source/data-handling-database-R.Rmd (hunk 2 above)
Comment (ThierryO) [inline source/data-handling-database-R.Rmd]: Note that you need to update line 80 and 81 as well

File: source/data-handling-database-R.Rmd [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+```
+
+For Windows users, the most important element is to know the so-called `DSN` (i.e. a registered Data Source Name). Actually, it is just the name of the database as it is known by your computer (and MS Access). The easiest way to check the `DSN` is to check the [*registered ODBC connections*](http://www.stata.com/support/faqs/data-management/configuring-odbc-win/) in the administrator tools menu. 
+
+For Dutch-speaking Windows 7 users: 
+
+    > Kies in het Configuratiescherm van Windows de optie Systeembeheer > Gegevensbronnen (ODBC). De optie Systeembeheer verschijnt in de categorie Systeem en onderhoud.
+
+You should see a list similar to the list underneath, with the names of the available DSN names enlisted:
+![odbc-connecties](../img/odbc_gegevensbron.png)
```
Comment (florisvdh) [inline source/data-handling-database-R.Rmd]: the png's don't render in my case

Comment (stijnvanhoey) [inline source/data-handling-database-R.Rmd]: my mistake, I added them to the repo now

**Pull request merged**